pandas/io/pickle.py

```diff
@@ -15,12 +16,18 @@ def to_pickle(obj, path):
     obj : any object
     path : string
         File path
+    compression : {'infer', 'gzip', 'bz2', 'xz', None}, default 'infer'
     """
-    with open(path, 'wb') as f:
+    inferred_compression = _get_inferred_compression(path, compression)
+    if inferred_compression:
```

Comment:
can just do ``_get_handle`` (it will open if necessary when no compression is specified)